deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -270,174 +264,178 @@ def _copy_all_manifest_files_from_old_hash_buckets(
     return materialize_result_list
 
 
-def _timed_merge(input: MergeInput) -> MergeResult:
-    def _materialize(
-        hash_bucket_index,
-        compacted_tables: List[pa.Table],
-    ) -> MaterializeResult:
-        compacted_table = pa.concat_tables(compacted_tables)
-        if input.compacted_file_content_type in DELIMITED_TEXT_CONTENT_TYPES:
-            # TODO (rkenmi): Investigate if we still need to convert this table to pandas DataFrame
-            # TODO (pdames): compare performance to pandas-native materialize path
-            df = compacted_table.to_pandas(split_blocks=True, self_destruct=True)
-            compacted_table = df
-        delta, stage_delta_time = timed_invocation(
-            input.deltacat_storage.stage_delta,
-            compacted_table,
-            input.write_to_partition,
-            max_records_per_entry=input.max_records_per_output_file,
-            content_type=input.compacted_file_content_type,
-            s3_table_writer_kwargs=input.s3_table_writer_kwargs,
+def _compact_tables(
+    input: MergeInput, dfe_list: List[List[DeltaFileEnvelope]], hb_idx: Optional[int]
+) -> Tuple[pa.Table, int]:
+    hb_prefix_log_statement = f"[Hash bucket index {hb_idx}]" if hb_idx else "[Local]"
```

Comment:
I felt it was a little misleading to have a default hash bucket index here, since we aren't actually doing any hashing of the PK columns for local DFEs. But to help simplify the code, it might be more suitable to document somewhere that we used a fixed `hb_idx` for local DFEs.